cmd/gophermart/cmd/root.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright © 2023 NAME HERE <EMAIL ADDRESS>
+*/
+package cmd
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/shipherman/gophermart/internal/clients"
+	"github.com/shipherman/gophermart/internal/db"
+	"github.com/shipherman/gophermart/internal/handlers"
+	gmiddw "github.com/shipherman/gophermart/internal/transport/middleware"
+	"github.com/shipherman/gophermart/internal/transport/routes"
+	"github.com/shipherman/gophermart/internal/transport/worker"
+
+	"github.com/caarlos0/env/v8"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/spf13/cobra"
+)
+
+type Options struct {
+	DSN     string `env:"DATABASE_URI"`
+	Accrual string `env:"ACCRUAL_SYSTEM_ADDRESS"`
+	Address string `env:"RUN_ADDRESS"`
+}
+
+var cfg Options
+
+// Init new logger;
+// To do
+// Make it general to gophermart app
+var logEntry = middleware.DefaultLogFormatter{Logger: log.New(os.Stdout, "", log.LstdFlags)}
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "gophermart",
+	Short: "A brief description of your application",
+	Long: `A longer description that spans multiple lines and likely contains
+examples and usage of using your application. For example:
+
+Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
+	// Uncomment the following line if your bare application
+	// has an action associated with it:
+	// Run: func(cmd *cobra.Command, args []string) { },
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	err := rootCmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+
+	dbclient := db.NewClient(cfg.DSN)
+	err = dbclient.Start()
+	if err != nil {
+		os.Exit(1)
+	}
+
+	// Set accruall address
+	clients.SetAccrualAddress(cfg.Accrual)
+
+	// put accrual worker here
+	aWorker := worker.New(dbclient)
+	wg := sync.WaitGroup{}
+
+	wg.Add(2)
+	go func(wg *sync.WaitGroup) {
+		go aWorker.Run(wg)
```

Comment:
горутина в горутине - не имеет смысла
можно вынести выше до вызова родительской горутины и логика работы никак не изменится
```suggestion
        go aWorker.Run(wg)
	go func(wg *sync.WaitGroup) {
		
```